syntax = "proto3";
package payment;

service PaymentService {
  rpc CreatePayment (PaymentRequest) returns (PaymentResponse);
  rpc DeletePayment (PaymentIdRequest) returns (PaymentResponse);
  rpc GetAllPayments (Empty) returns (PaymentList);
}

message PaymentRequest {
  int32 user_id = 1;
  int32 booking_id = 2;
  float amount = 3;
  string method = 4;
  string status = 5;
}

message PaymentIdRequest {
  int32 payment_id = 1;
}

message PaymentResponse {
  string message = 1;
  bool success = 2;
}

message Empty {}

message Payment {
  int32 id = 1;
  int32 user_id = 2;
  int32 booking_id = 3;
  float amount = 4;
  string method = 5;
  string status = 6;
  string timestamp = 7;
}

message PaymentList {
  repeated Payment payments = 1;
}
